packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2153,8 +2228,9 @@ function commitMutationEffects_begin(root: FiberRoot, lanes: Lanes) {
     if (deletions !== null) {
       for (let i = 0; i < deletions.length; i++) {
         const childToDelete = deletions[i];
+        const outermostHiddenFiber = findOutermostHiddenFiber(childToDelete);
```

Comment:
This searches up to the root for each fiber in the deletion list. I don't know if there's a way to avoid this. We need to somehow know where we are in the tree before we start descending.